### PR TITLE
Feature/content highlight

### PR DESCRIPTION
**Pull request by marcelmoosbrugger**
__tasks:__

- [x] implement highlight section in content-bundle
- [ ] gather feedback for my changes

__depending prs:__
https://github.com/massiveart/husky/pull/538

__informations:__

| q                | a
| ---------------- | ---
| Fixed tickets    | None
| BC breaks        | None
| Documentation PR | None

### CODE REVIEW

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-content.js [hunk 1]
```diff
@@ -86,7 +86,8 @@ define(function() {
          *
          * @param {Object|Boolean|Function} layout The layout object or true for default values.
          *        If a function, gets called and takes the return value to work with.
-         * @param {Boolean} [layout.changeNothing] If true the layout as it is won't be touched.
+         * @param {Boolean} [layout.extendExisting] If true true, the existing layout stays as it is and only the spcified
```
Comment (nnatter) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-content.js]: @marcelmoosbrugger do we need 2 "true" here?

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-content.js [hunk 2]
```diff
@@ -136,17 +143,18 @@ define(function() {
          * Handles the navigation part of the layout object.
          *
          * @param {Object} navigation The navigation config object.
+         * @param {Boolean} defaults Iff true default actions will be executed
```
Comment (nnatter) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-content.js]: @marcelmoosbrugger *if

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-content.js [hunk 3]
```diff
@@ -155,37 +163,45 @@ define(function() {
          * Handles the content part of the layout object.
          *
          * @param {Object} content The content config object.
+         * @param {Boolean} defaults Iff true default actions will be executed
```
Comment (nnatter) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-content.js]: @marcelmoosbrugger *if

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-content.js [hunk 4]
```diff
@@ -155,37 +163,45 @@ define(function() {
...
+            if (defaults === true || typeof content.shrinkable !== 'undefined') {
+                this.sandbox.emit('sulu.app.toggle-shrinker', !!content.shrinkable);
+            }
         },
 
         /**
          * Handles the sidebar part of the layout object.
          *
          * @param {Object} sidebar The sidebar config object. If false sidebar gets hidden.
+         * @param {Boolean} defaults Iff true default actions will be executed
```
Comment (nnatter) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-content.js]: @marcelmoosbrugger *if

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-content.js [hunk 5]
```diff
@@ -136,17 +143,18 @@ define(function() {
          * Handles the navigation part of the layout object.
          *
          * @param {Object} navigation The navigation config object.
+         * @param {Boolean} defaults Iff true default actions will be executed
          */
-        handleLayoutNavigation = function(navigation) {
+        handleLayoutNavigation = function(navigation, defaults) {
```
Comment (nnatter) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-content.js]: i think "useDefaukts"/"applyDefaults" would be a better name for the second parameter

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-content.js [hunk 6]
```diff
@@ -155,37 +163,45 @@ define(function() {
          * Handles the content part of the layout object.
          *
          * @param {Object} content The content config object.
+         * @param {Boolean} defaults Iff true default actions will be executed
          */
-        handleLayoutContent = function(content) {
+        handleLayoutContent = function(content, defaults) {
```
Comment (nnatter) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-content.js]: i think "useDefaukts"/"applyDefaults" would be a better name for the second parameter

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-content.js [hunk 7]
```diff
@@ -155,37 +163,45 @@ define(function() {
...
         },
 
         /**
          * Handles the sidebar part of the layout object.
          *
          * @param {Object} sidebar The sidebar config object. If false sidebar gets hidden.
+         * @param {Boolean} defaults Iff true default actions will be executed
          */
-        handleLayoutSidebar = function(sidebar) {
+        handleLayoutSidebar = function(sidebar, defaults) {
```
Comment (nnatter) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-content.js]: i think "useDefaukts"/"applyDefaults" would be a better name for the second parameter

File: src/Sulu/Bundle/ContentBundle/Resources/views/Template/content-types/text_line.html.twig [hunk 8]
```diff
@@ -1,7 +1,14 @@
+{% set bold = false %}
```
Comment (nnatter) [inline src/Sulu/Bundle/ContentBundle/Resources/views/Template/content-types/text_line.html.twig]: @marcelmoosbrugger i do not have the qualifications to review this

File: src/Sulu/Bundle/ContentBundle/Resources/views/Template/content.html.twig [hunk 9]
```diff
@@ -1,24 +1,43 @@
-<form id="content-form" class="grid">
```
Comment (nnatter) [inline src/Sulu/Bundle/ContentBundle/Resources/views/Template/content.html.twig]: @marcelmoosbrugger i do not have the qualifications to review this

File: src/Sulu/Bundle/ContentBundle/Resources/views/Template/macros/property.html.twig [hunk 10]
```diff
@@ -0,0 +1,17 @@
+{# get params for property #}
```
Comment (nnatter) [inline src/Sulu/Bundle/ContentBundle/Resources/views/Template/macros/property.html.twig]: @marcelmoosbrugger i do not have the qualifications to review this

File: src/Sulu/Bundle/ContentBundle/Resources/views/Template/content-types/text_line.html.twig (hunk 8 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContentBundle/Resources/views/Template/content-types/text_line.html.twig]: hmm i think you can only do this: 
```
{% if params.bold is defined %}
{% set bold = params.bold %}
{% else %}
{% set bold = false %}
{% endif %}
```

if you also define a defaultParam in the content type with `bold = false` then you can get rid of the `if is defined `

File: src/Sulu/Bundle/ContentBundle/Resources/views/Template/content.html.twig (hunk 9 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContentBundle/Resources/views/Template/content.html.twig]: if we go to the section you can get rid of this ... i think

File: src/Sulu/Bundle/ContentBundle/Resources/views/Template/macros/property.html.twig (hunk 10 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContentBundle/Resources/views/Template/macros/property.html.twig]: thats ok!

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-content.js (hunk 1 above)
Comment (marcelmoosbrugger) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-content.js]: @nnatter fixed

**Pull request merged**